extensions/cornerstone-3d/src/Viewport/Overlays/ViewportOverlay.tsx

```diff
@@ -219,10 +194,66 @@ function CornerstoneOverlay({
   );
 }
 
-CornerstoneOverlay.propTypes = {
+function _getInstanceNumberFromStack(viewportData, imageIndex) {
+  const imageIds = viewportData.imageIds;
+  const imageId = imageIds[imageIndex];
+
+  if (!imageId) {
+    return;
+  }
+
+  const generalImageModule = metaData.get('generalImageModule', imageId) || {};
+  const { instanceNumber } = generalImageModule;
+
+  const stackSize = imageIds.length;
+
+  if (stackSize <= 1) {
+    return;
+  }
+
+  return parseInt(instanceNumber);
+}
+
+function _getInstanceNumberFromVolume(viewportData, imageIndex, viewportIndex) {
```

Comment:
I would give a comment about why this only works in the acquistion direction, just to be super clear to the reader.